[Integrations] Alerting Rule Template common page

alaudazzi opened the pull request:
<!--
Thank you for contributing to the Elastic Docs! 🎉
Use this template to help us efficiently review your contribution.
-->

## Summary

This PR adds a new Alerting Rule Templates page that should serve as a centralized doc page with an explanation of what the alert rule templates are and how to use them. This page is going to be referenced from the individual integration pages.

The current location of the page will be changed and made more visible in a further iteration. 


Relates to https://github.com/elastic/docs-content/issues/3678#issuecomment-3561707267. 


## Generative AI disclosure
<!--
To help us ensure compliance with the Elastic open source and documentation guidelines, please answer the following:
-->
1. Did you use a generative AI (GenAI) tool to assist in creating this contribution?
- [x] Yes  
- [ ] No  

Cursor with gpt5

daniela-elastic commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,50 @@
+---
+applies_to:
+  stack: ga 9.3
```
I believe the correct version is 9.2.1 @muthu-mps could you please verify this is correct?

daniela-elastic commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,50 @@
+
+# Alerting Rule Templates [alerting-rule-templates]
+
+Alerting rule templates are out-of-the-box, preconfigured rule definitions maintained by Elastic integration authors. They help you start monitoring in minutes—no queries to write, no thresholds to figure out—by providing curated {{esql}} queries, sensible defaults, and recommended thresholds tailored to each integration. Templates are available from an integration’s Assets and open a prefilled r ...
```
Do we want to mention that while  everything is preconfigured but you can customize anything if you wish. Or we want to keep the mention of customization below only?

daniela-elastic commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,50 @@
+1. Select a template to open a prefilled Create rule form.
+1. Review and (optionally) customize the prefilled settings, then save and enable the rule.
+
+When you click a template, you get a prefilled **Create Rules** form. You can use the template to create your own custom alerting rule by adjusting values, setting up connectors, and defining rule actions.
```
"click a template" or "click on a template" ?

daniela-elastic reviewed (approved):
Made minor comments and also question on the correct stack version. Approving in principal but please review the comments.

tommyers-elastic commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,49 @@
+
+## How to use the Alerting Rule Templates
+
+Alerting rule templates come with recommended, pre-populated values. To use them:
```
these instructions will become out of date v quickly (by 9.3). let's open a ticket to remind us to update them with the new UI flow for the next release.

alaudazzi commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,49 @@
+
+# Alerting Rule Templates [alerting-rule-templates]
+
+Alerting rule templates are out-of-the-box, preconfigured rule definitions maintained by Elastic integration authors. They help you start monitoring in minutes—no queries to write, no thresholds to figure out—by providing curated {{esql}} queries, sensible defaults, and recommended thresholds tailored to each integration. Templates are available from an integration’s Assets and open a prefilled r ...
```
Rephrased.

alaudazzi commented:
@tommyers-elastic @daniela-elastic 
Your comments have been addressed. Please have a final review before I can merge.

alaudazzi commented:
@tommyers-elastic I added the following note as you suggested:

> Although the alerts can be used as provided, threshold values should always be evaluated in the context of your specific environment. Applying the predefined thresholds without adjustment may result in an excessive number of alerts.

alaudazzi commented:
@daniela-elastic 
I addressed your suggestions as follows:

> Should we add a hyperlink the first time we mention "Elastic integrations" so that users who came straight to this page know what integrations we are talking about

=> done

> Should we mention that while the template is managed by Elastic (and it may change in the future), once the customer creates an alert out of it, it's theirs and Elastic doesn't change it even if the template changes. That way customers who are used to the integrations being managed (the dashboard is updated automatically unless you clone it) don't get the wrong impression that the alerts will change as well.

=> done
I slightly rephrased as follows:
`Although these templates are managed by Elastic, any alert created from them is owned by the customer and will not be modified by Elastic, even if the templates change.`


> Should we mention that if an integration doesn't show alert templates in the asset tab, that doesn't mean that there's necessarily a bug but some integrations don't yet have alerts

=> This note should already clarify this point

`{note} You can find the Alerting rule template option only when the integration adds template support for alerting rules.`

> In "Select a template to open a prefilled Create rule form. [...] setting up connectors, and defining rule actions." should we define what we mean by connectors. In this case I believe these are connectors to eg email, etc that will send the alert info to the channel where you normally receive your alerts

@muthu-mps can you help clarify what these connectors are?

daniela-elastic commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,69 @@
+Although these templates are managed by Elastic, any alert created from them is owned by the customer and will not be modified by Elastic, even if the templates change.
+
+:::{important}
+Although the alerts can be used as provided, threshold values should always be evaluated in the context of your specific environment. Applying the predefined thresholds without adjustment may result in an excessive number of alerts.
```
Depending on how you adjust the thresholds, you may have too many alerts or your alerts may not trigger when you would expect them to trigger

theletterf commented on reference/fleet/alerting-rule-templates.md:
```diff
@@ -0,0 +1,69 @@
+
+Templates help you start monitoring in minutes by providing curated {{esql}} queries and recommended thresholds tailored to each integration. 
+
+Once the integration is installed, these templates are automatically available in Kibana's alerting interface with a prefilled rule creation form that you can tailor to your needs.
```
```suggestion
After the integration is installed, these templates are automatically available in Kibana's alerting interface with a prefilled rule creation form that you can adapt to your needs.
```
